### ISSUE TITLE
Follow up to #2 :  Resolve the conflict between right-click and the Outliner

### ISSUE THREAD

**Issue by LeapSurge** [ReticentEclectic/checkbox-style-menu#3]
Regarding the issue mentioned in #2 , I reviewed both the Outliner and this project’s code. **The problem caused by using right-click is not the right-click itself, but the way the menu item selection is handled.**

Currently, the `handleStyleSelection` logic is triggered via a bound `mouseup` event, which was originally designed to match the previous long-press interaction.

```typescript
private setupEventListeners(view: EditorView) {
  //...
  // Platform-specific interaction handling
  if (Platform.isMobile) {
      this.setupTouchHandling(view, signal);
  } else {
      // Desktop: simple click handling
      this.menuElement.addEventListener('mouseup', (e: MouseEvent) => {
          const li = (e.target as HTMLElement).closest('li');
          if (li) {
              e.stopPropagation();
              e.preventDefault();
              this.handleStyleSelection(view, li);
          }
      }, { signal });
  //...
}
``` 

After switching to right-click as the trigger, the right-click itself initiates a complete Outliner drag operation and opens the menu, which is expected and works correctly. **The issue occurs when clicking a menu item**: the Outliner is waiting for a `mouseup` to terminate the drag, but at that moment the `mouseup` event is prevented. As a result, the Outliner remains in a dragging state, continuously waiting for the `mouseup`.

To prevent this situation, we should use different event listeners depending on the trigger method. A check for the trigger method should be added before **line 585**.

```typescript
//...
const event =this.plugin.settings.triggerMethod === "right-click"
  ? "click"
  : "mouseup";
this.menuElement.addEventListener(event,(e: MouseEvent) => {
// ...
```

**Comment by ReticentEclectic** [comment]
Thank you for reviewing the code. That's a great catch on the `mouseup` listener from the previous long-press trigger method. Your suggested fix worked perfectly to listen specifically for `click` for the right-click trigger and `mouseup` for the long-press trigger. I went through and made sure each trigger method listens for the appropriate event.

Additionally, I noticed that the overlay was preventing Outliner from receiving the `mouseup` event to complete its drag. So I modified the overlay's listeners to allow `mouseup` to propagate through, keeping it Outliner-friendly.

Thank you for the feedback! Please let me know if you notice any other conflicts or bugs.
